exercise3/src/camera_demo/CMakeLists.txt

```diff
@@ -0,0 +1,80 @@
+cmake_minimum_required(VERSION 3.8)
+project(camera_demo)
+
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
+
+# find dependencies
+find_package(ament_cmake REQUIRED)
+# uncomment the following section in order to fill in
+# further dependencies manually.
+# find_package(<dependency> REQUIRED)
+
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+
+# Integrated cactus-rt dependencies for the workshop
+add_subdirectory(
+  ${CMAKE_CURRENT_SOURCE_DIR}/../../../vendor/cactus-rt-integrated
+  ${CMAKE_CURRENT_BINARY_DIR}/cactus-rt-build
+)
+
+include_directories(
+  include
+  ${OpenCV_INCLUDE_DIRS}
+)
+
+add_library(publisher_node src/publisher_node.cpp)
+ament_target_dependencies(publisher_node
+  rclcpp
+  sensor_msgs
+)
+target_link_libraries(publisher_node
+  cactus_rt
+)
+
+add_library(sensor_node src/sensor_node.cpp)
+ament_target_dependencies(sensor_node
+  rclcpp
+  sensor_msgs
+)
+target_link_libraries(sensor_node
+  cactus_rt
+)
+
+add_library(actuation_node src/actuation_node.cpp)
+ament_target_dependencies(actuation_node
+  rclcpp
+  sensor_msgs
+)
+target_link_libraries(actuation_node
+  cactus_rt
+)
+
+add_executable(camera_demo src/camera_demo.cpp)
+ament_target_dependencies(camera_demo
+  rclcpp
+)
+target_link_libraries(camera_demo
+  publisher_node sensor_node actuation_node cactus_rt
```

Comment:
I don't understand why the nodes are defined as libraries? Why not simply do 

```
add_executable(camera_demo
 src/camera_demo.cpp
 src/sensor_node.cpp
 ...
)
```

Reply:
That makes recompilation faster to spin out nodes that aren't editted

Reply:
If you don't change the individual cpp files they shouldn't be recompiled as they are separate compilation units already.

Reply:
Updated

Reply:
I merged it already. If you can, create a new PR against main. Thanks!